src/test/java/org/candlepin/model/ConsumerCloudDataTest.java

```diff
@@ -93,6 +93,26 @@ public void testSetCloudInstanceId() {
         assertEquals("cloudInstanceId exceeds the max length", exception.getMessage());
     }
 
+    @Test
+    public void testAddCloudOfferingIds() {
```

Comment:
Let's split this up into distinct tests for the expected cases (single, multiple for both the array and collection input), and then our null, empty, and malformed cases.